app/controllers/cloud_volume_controller.rb

```diff
@@ -129,6 +129,22 @@ def cloud_volume_form_fields
     }
   end
 
+  def storage_manager_form_fields
+    assert_privileges("cloud_volume_new")
+    ems = find_by_id_filtered(ExtManagementSystem, params[:id])
+
+    form_fields = { :emstype => ems.emstype }
+
+    case ems.type
+    when "ManageIQ::Providers::Amazon::StorageManager::Ebs"
+      form_fields[:availability_zone_choices] = ems.availability_zones.each_with_object([]) { |az, choices| choices << az.name }.sort
+    when "ManageIQ::Providers::StorageManager::CinderManager"
+      form_fields[:cloud_tenant_choices] = ems.parent_manager.cloud_tenants.each_with_object({}) { |tenant, choices| choices[tenant.name] = tenant.id }
+    end
```

Comment:
@gberginc I'm looking for a possible REST API to replace the above.

With an API like that:
```
/api/providers/<id>?attributes=type,parent_manager.availability_zones,parent_manager.cloud_tenants
```
you can retrieve all 3 form fields in one go.

But just wondering -- why is it that for `availability_zones` you don't need a `parent_manager` and for `cloud_tenants` you do?